# Local equivalent (this release lacks the riscv64/thread-pointer lines) of:
# Upstream-Commit: https://github.com/llvm/llvm-project/commit/62836024df450d2d3afbeb7f61ec8bdab3060ed6
diff --git a/llvm/BUILD.bazel b/llvm/BUILD.bazel
--- a/llvm/BUILD.bazel
+++ b/llvm/BUILD.bazel
@@ -33,6 +33,42 @@ exports_files([
     "include/llvm/Option/OptParser.td",
 ])
 
+config_setting(
+    name = "windows",
+    visibility = ["//visibility:public"],
+    constraint_values = ["@platforms//os:windows"],
+)
+
+config_setting(
+    name = "darwin_arm64",
+    visibility = ["//visibility:public"],
+    constraint_values = ["@platforms//os:macos", "@platforms//cpu:arm64"],
+)
+
+config_setting(
+    name = "darwin_x86_64",
+    visibility = ["//visibility:public"],
+    constraint_values = ["@platforms//os:macos", "@platforms//cpu:x86_64"],
+)
+
+config_setting(
+    name = "linux_aarch64",
+    visibility = ["//visibility:public"],
+    constraint_values = ["@platforms//os:linux", "@platforms//cpu:aarch64"],
+)
+
+config_setting(
+    name = "linux_ppc64le",
+    visibility = ["//visibility:public"],
+    constraint_values = ["@platforms//os:linux", "@platforms//cpu:ppc"],
+)
+
+config_setting(
+    name = "linux_s390x",
+    visibility = ["//visibility:public"],
+    constraint_values = ["@platforms//os:linux", "@platforms//cpu:s390x"],
+)
+
 # It may be tempting to add compiler flags here, but that should be avoided.
 # The necessary warnings and other compile flags should be provided by the
 # toolchain or the `.bazelrc` file. This is just a workaround until we have a
diff --git a/llvm/config.bzl b/llvm/config.bzl
--- a/llvm/config.bzl
+++ b/llvm/config.bzl
@@ -80,20 +80,20 @@ win32_defines = [
 # TODO: We should switch to platforms-based config settings to make this easier
 # to express.
 os_defines = select({
-    "@bazel_tools//src/conditions:windows": win32_defines,
-    "@bazel_tools//src/conditions:darwin": macos_defines,
-    "@bazel_tools//src/conditions:freebsd": posix_defines,
+    "@platforms//os:windows": win32_defines,
+    "@platforms//os:macos": macos_defines,
+    "@platforms//os:freebsd": posix_defines,
     "//conditions:default": linux_defines,
 })
 
 # TODO: We should split out host vs. target here.
 llvm_config_defines = os_defines + select({
-    "@bazel_tools//src/conditions:windows": native_arch_defines("X86", "x86_64-pc-win32"),
-    "@bazel_tools//src/conditions:darwin_arm64": native_arch_defines("AArch64", "arm64-apple-darwin"),
-    "@bazel_tools//src/conditions:darwin_x86_64": native_arch_defines("X86", "x86_64-unknown-darwin"),
-    "@bazel_tools//src/conditions:linux_aarch64": native_arch_defines("AArch64", "aarch64-unknown-linux-gnu"),
-    "@bazel_tools//src/conditions:linux_ppc64le": native_arch_defines("PowerPC", "powerpc64le-unknown-linux-gnu"),
-    "@bazel_tools//src/conditions:linux_s390x": native_arch_defines("SystemZ", "systemz-unknown-linux_gnu"),
+    "//llvm:windows": native_arch_defines("X86", "x86_64-pc-win32"),
+    "//llvm:darwin_arm64": native_arch_defines("AArch64", "arm64-apple-darwin"),
+    "//llvm:darwin_x86_64": native_arch_defines("X86", "x86_64-unknown-darwin"),
+    "//llvm:linux_aarch64": native_arch_defines("AArch64", "aarch64-unknown-linux-gnu"),
+    "//llvm:linux_ppc64le": native_arch_defines("PowerPC", "powerpc64le-unknown-linux-gnu"),
+    "//llvm:linux_s390x": native_arch_defines("SystemZ", "systemz-unknown-linux_gnu"),
     "//conditions:default": native_arch_defines("X86", "x86_64-unknown-linux-gnu"),
 }) + [
     "LLVM_VERSION_MAJOR={}".format(LLVM_VERSION_MAJOR),
